modules/api/src/main/java/net/countercraft/movecraft/TrackedLocation.java

```diff
@@ -0,0 +1,57 @@
+package net.countercraft.movecraft;
+
+import org.bukkit.World;
+import org.jetbrains.annotations.NotNull;
+
+public class TrackedLocation {
+    private int posX;
+    private int posY;
+    private int posZ;
+    private MovecraftLocation origin;
+
+    /**
+     * Creates a new TrackedLocation instance which tracks a location about an origin.
+     * @param origin The origin point at which the tracked location rotates about.
+     * @param location The absolute location to track.
+     */
+    public TrackedLocation(@NotNull MovecraftLocation origin, @NotNull MovecraftLocation location) {
```

Comment:
Rather than store the origin, the `TrackedLocation` should store a reference to the `Craft` it is part of here.

Reply:
I'm not sure the best thing to use as an origin, `Craft`s currently don't track their origin point but perhaps we should?  The hitbox midpoint will change as a craft is damaged, so that's not a good option either.

Reply:
The only reason why I need an origin point is because of rotation but because I realize now that crafts rotate at a different point for any kind of rotation. Perhaps it might be appropriate to change the origin point with every rotation? I'm not sure how to handle this without a listener. But if this were to be the case, then I might have to just store the absolute value of the craft and require the rotations to also have a rotation point. RotationTask has one iirc so I could just use that.

Reply:
We do need to make sure we rotate the `TrackedLocation`s properly on rotation, but that's a separate origin as the one used here.

The `Craft` `origin` is a coordinate in the hitbox that we will be choosing at detection (likely just the midpoint of the detected hitbox), then all `TrackedLocation`s only keep track of their offset from that.  That way on every movement, you don't need to do any processing to the `TrackedLocation`s.  They do need to be changed on rotation, but that happens less often.  When resolving the absolute world position of a `TrackedLocation`, it would take the `Craft`'s `origin` and add it's own `offset`.

While this method has no performance improvement over the method of always translating IF the `TrackedLocation`s have their absolute world position computed every movement, it does have a performance benefit if the `TrackedLocation`s are only queried for their absolute world position every once in a while.